gobblin-metastore/src/main/java/org/apache/gobblin/metastore/DatasetStateStore.java

```diff
@@ -19,16 +19,30 @@
 
 import java.io.IOException;
 import java.util.Collection;
+import java.util.List;
 import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
+import com.google.common.base.Strings;
 import com.typesafe.config.Config;
 
+import org.apache.gobblin.configuration.ConfigurationKeys;
 import org.apache.gobblin.configuration.State;
+import org.apache.gobblin.metastore.metadata.DatasetStateStoreEntryMetadata;
+import org.apache.gobblin.metastore.predicates.StateStorePredicate;
+import org.apache.gobblin.util.ClassAliasResolver;
+import org.apache.gobblin.util.ConfigUtils;
+
+import lombok.Getter;
+
 
 public interface DatasetStateStore<T extends State> extends StateStore<T> {
   String DATASET_STATE_STORE_TABLE_SUFFIX = ".jst";
   String CURRENT_DATASET_STATE_FILE_SUFFIX = "current";
 
+  Pattern TABLE_NAME_PARSER_PATTERN = Pattern.compile("^(?:(.+)-)?([^-]+)\\.jst$");
```

Comment:
Added a step to sanitize the job id before writing to the store. This is safe, because the only file we ever read has the id `current`, which is not changed.